checkers/typeDefFirst_checker.go

```diff
@@ -30,44 +30,48 @@ func (r rec) Method() {}
 
 type typeDefFirstChecker struct {
 	astwalk.WalkHandler
-	ctx *linter.CheckerContext
+	ctx          *linter.CheckerContext
+	trackedTypes map[string]bool
```

Comment:
nit: why do we need this parameter? Feels like it can be the second argument of the function `walkDecl()`, it makes function purer, idk.

Reply:
But it's not a function, it's a method.
And trackedTypes is not a parameter, it's a struct field.
Checkers are generally can't be used without a state, they usually have at least some kind of state.

Reply:
I'm ok with a field